api/client/alpn_conn_upgrade.go

```diff
@@ -271,19 +261,10 @@ func upgradeConnThroughWebAPI(conn net.Conn, api url.URL, alpnUpgradeType string
 
 	// Handle WebSocket.
 	logger := slog.With("hostname", api.Host)
-	if resp.Header.Get(constants.WebAPIConnUpgradeHeader) == constants.WebAPIConnUpgradeTypeWebSocket {
-		if err := checkWebSocketUpgradeResponse(resp, alpnUpgradeType, challengeKey); err != nil {
-			return nil, trace.Wrap(err)
-		}
-
-		logger.DebugContext(req.Context(), "Performing ALPN WebSocket connection upgrade.")
-		return newWebSocketALPNClientConn(conn), nil
+	if err := checkWebSocketUpgradeResponse(resp, alpnUpgradeType, challengeKey); err != nil {
```

Comment:
`resp.Header.Get(constants.WebAPIConnUpgradeHeader) == constants.WebAPIConnUpgradeTypeWebSocket` check is moved inside this func